src/gui/mainwindow.cpp

```diff
@@ -325,7 +327,9 @@ void MainWindow::restoreWindowGeometry()
 #endif
 
 	QSettings settings("nvim-qt", "window-geometry");
-	restoreGeometry(settings.value("window_geometry").toByteArray());
+	if (settings.value("restore_window_geometry", true).toBool()) {
+		restoreGeometry(settings.value("window_geometry").toByteArray());
+	}
 	restoreState(settings.value("window_state").toByteArray());
```

Comment:
I think we should also avoid running `restoreState` in this case too.

Something like:
```C++
	if (!settings.value("restore_window_geometry", true).toBool()) {
		return;
	}
	
	restoreGeometry(settings.value("window_geometry").toByteArray());
	restoreState(settings.value("window_state").toByteArray());
```